sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -145,17 +148,18 @@ az_add_header_to_curl_list(az_pair header, struct curl_slist ** p_list, az_span
  * @param p_headers list of headers in curl specific list
  * @return az_result
  */
-AZ_NODISCARD az_result
-az_build_headers(az_http_request * p_request, struct curl_slist ** p_headers) {
+static AZ_NODISCARD az_result _az_http_client_curl_build_headers(
+    az_http_request * p_request,
+    struct curl_slist ** p_headers) {
   AZ_CONTRACT_ARG_NOT_NULL(p_request);
 
   az_pair header;
   for (int32_t offset = 0;
        offset < (int32_t)(az_span_length(p_request->_internal.headers) / sizeof(az_pair));
```

Comment:
Don't we have functions for this now? The CURL code should NOT be accessing internal fields directly.